xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/AttachmentsPane.java

```diff
@@ -309,6 +320,32 @@ public boolean attachmentExistsByFileName(String attachmentName)
         return true;
     }
 
+    /**
+     * Check if the attachment exists, regardless of the page where it is displayed.
+     *
+     * @param attachmentName the name of the searched attachment
+     * @return whether the attachments exists or not in the list of all attachments
+     */
+    public boolean attachmentExistsByFileName(String attachmentName)
+    {
+        while (!this.attachmentIsDisplayedByFileName(attachmentName)
+            && attachmentsLivetable.getNextPageElement().get() != null) {
+            attachmentsLivetable.getNextPageElement().get().click();
+        }
+        return this.attachmentIsDisplayedByFileName(attachmentName);
```

Comment:
The common practice for checking if a live table has an entry (that may not be on the first page) is to **filter** the live table. In this case, couldn't you filter the attachments live table by the attachment file name and then check the result?